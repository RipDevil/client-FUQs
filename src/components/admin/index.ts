export * from './grips/users';
export * from './grips/fuqs';
export * from './portal';
