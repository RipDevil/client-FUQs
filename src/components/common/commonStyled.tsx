import styled from 'styled-components';

export const StyledSpinner = {
  SpinnerDiv: styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    z-index: 1000;
    margin: -50px 0px 0px -50px;
    max-width: 140px;
  `,
  BackgroundDiv: styled.div`
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 999;
    background-color: rgb(255, 255, 255);
  `,
  TextDiv: styled.div`
    text-align: center;
    margin-top: 1rem;
  `,
};
