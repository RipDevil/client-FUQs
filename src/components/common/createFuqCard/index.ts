export { CreateFuqCard } from './create-fuq-card';
