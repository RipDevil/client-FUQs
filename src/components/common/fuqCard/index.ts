export { FuqCard } from './fuq-card';
