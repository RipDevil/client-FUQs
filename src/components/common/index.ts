export { Credits } from './credits';
export { FuqCard } from './fuqCard';
export { Spinner } from './spinner';
