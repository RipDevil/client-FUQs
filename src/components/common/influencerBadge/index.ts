export { InfluencerBadge } from './influencer-badge';
