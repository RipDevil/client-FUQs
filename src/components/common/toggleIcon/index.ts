export { ToggleIcon } from './toggle-icon';
