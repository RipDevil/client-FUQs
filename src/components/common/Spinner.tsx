import * as React from 'react';
import { StyledSpinner as S } from './commonStyled';

export interface SpinnerProps {
  text?: string;
  transparent?: boolean;
}

const Spinner: React.FC<SpinnerProps> = ({
  text = '',
  transparent = false, // TODO: implement this stuff
}) => {
  return (
    <>
      <S.BackgroundDiv />
      <S.SpinnerDiv>
        <div>
          <i className='fas fa-10x fa-spin fa-cog' />
        </div>
        {text && (
          <S.TextDiv>
            <h3>{text}</h3>
          </S.TextDiv>
        )}
      </S.SpinnerDiv>
    </>
  );
};

export default Spinner;
