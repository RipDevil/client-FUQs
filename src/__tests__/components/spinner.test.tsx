import React from 'react';
import { render } from '@testing-library/react';
import { Spinner } from 'components/common';

describe('Spinner tests', () => {
  it('Should be rendered not transparent and without text', () => {
    const { container } = render(<Spinner />);

    expect(container).toMatchSnapshot();
  });

  it('Should be rendered transparent', () => {
    const { container } = render(<Spinner transparent={true} />);

    expect(container).toMatchSnapshot();
  });

  it('Should be rendered with text', () => {
    const { getByText } = render(<Spinner text={'Some text'} />);
    const element = getByText('Some text');
    expect(element).toBeInTheDocument();

    expect(element).toMatchSnapshot();
  });
});
