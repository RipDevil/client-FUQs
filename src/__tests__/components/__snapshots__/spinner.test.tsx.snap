// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner tests Should be rendered not transparent and without text 1`] = `
.c1 {
  position: fixed;
  top: 50%;
  left: 50%;
  z-index: 1000;
  margin: -50px 0px 0px -50px;
  max-width: 140px;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 999;
  background-color: rgba(253,253,253,1);
}

<div>
  <div
    class="c0"
  />
  <div
    class="c1"
  >
    <div>
      <i
        class="fas fa-10x fa-spin fa-cog"
      />
    </div>
    0
  </div>
</div>
`;

exports[`Spinner tests Should be rendered transparent 1`] = `
.c1 {
  position: fixed;
  top: 50%;
  left: 50%;
  z-index: 1000;
  margin: -50px 0px 0px -50px;
  max-width: 140px;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 999;
  background-color: rgba(253,253,253,0.3);
}

<div>
  <div
    class="c0"
  />
  <div
    class="c1"
  >
    <div>
      <i
        class="fas fa-10x fa-spin fa-cog"
      />
    </div>
    0
  </div>
</div>
`;

exports[`Spinner tests Should be rendered with text 1`] = `
<h3
  class="ant-typography ant-typography-danger"
>
  Some text
</h3>
`;
